File: pset6/credit/credit.py
# Function that checks the card
def check_card(number):
    # Holds the sum of numbers
    soma = 0
    # Holds the number of digits
    count = len(number)
    # Reverse the number so we can work with it
    number = number[::-1]
    # iterates through number string converting digit to integer and working with it
    for index, digit in enumerate(number):
        digit = int(digit)
        if ((index + 1) % 2 == 0):
            digit *= 2
            if digit > 9:
                digit -= 9
        soma += digit
    if soma % 10 == 0:
        return 1
    else:
        return 0
# Main


def main():
    number = input('Number: ')
    length = len(number)
    # Gets 2 first digits and converts to int
    company = int(number[0:2:])
    if (check_card(number) == 1 and length > 13):
        if (company > 50 and company < 56 and length == 16):
            print('MASTERCARD')
        elif (company == 34 or company == 37 and length == 15):
            print('AMEX')
        elif (company / 10 == 4 and length == 13 or length == 16 or length == 19):
            print('VISA')
    else:
        print('INVALID')


if __name__ == "__main__":
    main()

File: pset1/credit.c
#include <stdio.h>
#include <cs50.h>

int check_card(long long number);

int main()
{
    int con=0;
    long long num=0;
    
    printf("Number:\n");
    num = get_long_long();
    

    long long temp = num;

    while(temp) //Counts digits
	{
	    con++;
	    temp /= 10;
	}

	//printf("%d digits\n",con);

	temp = num;

	while (temp > 100LL) //Gets 2 First digits
    {
        temp = temp / 10LL;
    }

    int company_id = temp;

    //printf("Comp ID %d \n",company_id);

    if (check_card(num) == 1 && con > 13)
    {
    	if (company_id > 50 && company_id < 56 && con == 16)
	    {
	        printf("MASTERCARD\n") ;
	    }
	    else if ((company_id == 34 || company_id == 37) && (con == 15)) 
	    {
	        printf("AMEX\n") ;
	    }
	    else if ((company_id / 10 == 4) && (con == 13 || con == 16 || con == 19))
	    {
	        printf("VISA\n") ;
	    }

    }else{
    	printf("INVALID\n");
    }

    return 0;
}

int check_card(long long number){
	int sum = 0, count=0;

	while(number) //Counts digits
	{
	    count++;
	    number /= 10;
	}

    for (int i = 1; i <= count; i++)
    {
        int digit = number % 10LL;

        if (i % 2 == 0)
        {
            digit *= 2;

            if (digit > 9)
            {
                digit -= 9;
            }
        }

        sum += digit;

        number /= 10LL;
    }

    if (sum % 10 == 0) return 1;
    else return 0;

}